src/model/MultiValueColumn.ts

```diff
@@ -0,0 +1,281 @@
+
+    return f();
+  }
+
+
+  getLabel(row: any, index: number) {
+    return '' + this.getValue(row, index);
+  }
+
+  getRaw(row: any, index: number) {
+    return this.accessor(row, index, this.id, this.desc, this.findMyRanker());
+  }
+
+  getValue(row: any, index: number) {
+    var v = this.getRaw(row, index);
+    return (v);
+  }
+
+  getColor(data: any) {
+
+    return this.colorScale(data);
+  }
+
+  calculateCellDimension() {
+
+    return (this.getWidth() / this.bins);
+  }
+
+  getxScale(data) {
+    this.xposScale
+      .domain([0, this.bins - 1])
+      .range([0, this.getWidth()]);
+    return this.xposScale(data);
+  }
+
+  getyScale(data, rowheight) {
+    this.yposScale
+      .domain([this.min, this.max])
+      .range([rowheight, 0]);
+    return this.yposScale(data);
+  }
+
+
+  getbinaryColor() {
+
+    return this.colorrange;
+  }
+
+  getthresholdValue() {
+
+    return this.threshold;
+  }
+
+  getVerticalBarHeight(data, rowheight) {
+
+    if (this.min < this.threshold) {
+      this.verticalBarScale
+        .domain([this.min, this.max])
+        .range([0, rowheight / 2]);
+
+      this.verticalBarHeight = (rowheight / 2 - this.verticalBarScale(data));
+
+    } else {
+      this.verticalBarScale
+        .domain([this.min, this.max])
+        .range([0, rowheight]);
+
+      this.verticalBarHeight = this.verticalBarScale(data);
+    }
+
+    return this.verticalBarHeight;
+
+  }
+
+  getyposVerticalBar(data, rowheight) {
+
+    if (this.min < this.threshold) {
+      this.ypositionVerticalBar = (data < this.threshold) ? (rowheight / 2) : rowheight / 2 - this.getVerticalBarHeight(data, rowheight);   // For positive and negative value
+    } else {
+      this.ypositionVerticalBar = rowheight - this.getVerticalBarHeight(data, rowheight);
+    }
+
+    return this.ypositionVerticalBar;
+  }
+
+  getboxPlotData(data) {
+
+    const minval_arr = Math.min.apply(Math, data);
+    const maxval_arr = Math.max.apply(Math, data);
+
+    data.sort(numSort);
+    const q1 = this.boxPlotScale(d3.quantile(data, 0.25));
+    const median = this.boxPlotScale(d3.quantile(data, 0.50));
+    const q3 = this.boxPlotScale(d3.quantile(data, 0.75));
+    const min_val = this.boxPlotScale(minval_arr);
+    const max_val = this.boxPlotScale(maxval_arr);
+    const boxdata = {min: min_val, median: median, q1: q1, q3: q3, max: max_val};
+
+    return (boxdata);
+
+  }
+
+  getUserSortBy() {
+    return this.sortBy;
+  }
+
+  setUserSortBy(rank) {
+
+    let ascending = false;
+
+    switch (rank) {
+      case 'min':
+        ascending = true;
+        break;
+      case 'max':
+        ascending = false;
+        break;
+      case 'mean':
+        ascending = true;
+        break;
+      case 'median':
+        ascending = false;
+        break;
+      case 'q1':
+        ascending = true;
+        break;
+      case 'q3':
+        ascending = false;
+        break;
+      default:
+        ascending = false;
```

Comment:
in the end, you need another way to tell the underlying ranking, that it should be sort by this column and if it is already sorted by this column that the sort criteria changed. Changing the sort criteria doesn't need to apply a toggle of ascendign vs descending